File: frontend/lib/models/question.dart
class Question {
  final int id;
  final int surveyId;
  final String text;
  final String type;
  final List<String>? options;
  final String? helpText;
  final int? minValue;
  final int? maxValue;

  Question({
    required this.id,
    required this.surveyId,
    required this.text,
    required this.type,
    this.options,
    this.helpText,
    this.minValue,
    this.maxValue,
  });

  factory Question.fromJson(Map<String, dynamic> json) {
    return Question(
      id: json['id'] as int,
      surveyId: json['survey_id'] as int,
      text: json['text'] as String,
      type: json['type'] as String,
      options: json['options'] != null 
          ? List<String>.from(json['options'] as List)
          : null,
      helpText: json['help_text'] as String?,
      minValue: json['min_value'] as int?,
      maxValue: json['max_value'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'survey_id': surveyId,
      'text': text,
      'type': type,
      'options': options,
      'help_text': helpText,
      'min_value': minValue,
      'max_value': maxValue,
    };
  }
}

File: src/models/Question.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Question = sequelize.define('Question', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
  },
  survey_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: 'surveys',
      key: 'id',
    },
  },
  question_text: {
    type: DataTypes.JSONB,
    allowNull: false,
  },
  question_type: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  is_required: {
    type: DataTypes.BOOLEAN,
    defaultValue: false,
  },
  options: {
    type: DataTypes.JSONB,
    allowNull: true,
  },
  min_value: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  max_value: {
    type: DataTypes.INTEGER,
    allowNull: true,
  },
  question_order: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  placeholder: {
    type: DataTypes.JSONB,
    allowNull: true,
  },
  help_text: {
    type: DataTypes.JSONB,
    allowNull: true,
  },
  question_route: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  flag: {
    type: DataTypes.STRING,
    allowNull: true,
  },
  weight: {
    type: DataTypes.DECIMAL,
    defaultValue: 1,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
  updated_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'questions',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
});

module.exports = Question;